qp_shotgun/qc_trim/qc_trim.py

```diff
@@ -0,0 +1,266 @@
+            raise ValueError('No run prefix matching this fwd read: %s'
+                             % fwd_fn)
+
+        if run_prefix in used_prefixes:
+            raise ValueError('This run prefix matches multiple fwd reads: '
+                             '%s' % run_prefix)
+
+        if rev_fp is None:
+            samples.append((run_prefix, sn_by_rp[run_prefix], fwd_fp, None))
+        else:
+            rev_fn = basename(rev_fp)
+            # if we have reverse reads, make sure the matching pair also
+            # matches the run prefix:
+            if not rev_fn.startswith(run_prefix):
+                raise ValueError('Reverse read does not match this run prefix.'
+                                 '\nRun prefix: %s\nForward read: %s\n'
+                                 'Reverse read: %s\n' %
+                                 (run_prefix, fwd_fn, rev_fn))
+
+            samples.append((run_prefix, sn_by_rp[run_prefix], fwd_fp,
+                            rev_fp))
+
+        used_prefixes.add(run_prefix)
+
+    return(samples)
+
+
+def _format_qc_trim_params(parameters):
+    params = []
+
+    for param in sorted(parameters):
+        value = parameters[param]
+        dash = '--'
+        if len(param) == 1:
+            dash = '-'
+        if value is 'True':
+            params.append('%s%s' % (dash, param))
+        elif value is 'False':
+            continue
+        elif value and value != 'default':
+            params.append('%s%s %s' % (dash, param, value))
+
+    param_string = ' '.join(params)
+
+    return(param_string)
+
+
+def generate_qc_trim_commands(forward_seqs, reverse_seqs, map_file,
+                              out_dir, parameters):
+    """Generates the QC_Trim commands
+
+    Parameters
+    ----------
+    forward_seqs : list of str
+        The list of forward seqs filepaths
+    reverse_seqs : list of str
+        The list of reverse seqs filepaths
+    map_file : str
+        The path to the mapping file
+    out_dir : str
+        The job output directory
+    parameters : dict
+        The command's parameters, keyed by parameter name
+
+    Returns
+    -------
+    cmds: list of str
+        The KneadData commands
+    samples: list of tup
+        list of 4-tuples with run prefix, sample name, fwd read fp, rev read fp
+
+    Notes
+    -----
+    Currently this is requiring matched pairs in the make_read_pairs_per_sample
+    step but implicitly allowing empty reverse reads in the actual command
+    generation. This behavior may allow support of situations with empty
+    reverse reads in some samples, for example after trimming and QC.
+    """
+    # we match filenames, samples, and run prefixes
+    samples = make_read_pairs_per_sample(forward_seqs, reverse_seqs, map_file)
+    cmds = []
+
+    param_string = _format_qc_trim_params(parameters)
+
+    for run_prefix, sample, f_fp, r_fp in samples:
+        cmds.append('atropos trim --threads 4 %s -o %s -p %s -pe1 %s -pe2 %s'
```

Comment:
please make `threads` a parameter too, like [this](https://github.com/qiita-spots/qp-shotgun/blob/master/qp_shotgun/humann2/__init__.py#L44)